### PR TITLE
Add Anomaly Detection Rules

### PR DESCRIPTION
**Pull request by otetard**
This work was initially done by @bgigodeaux in #311. I just applied the new organization.

### CODE REVIEW

File: docs/xdr/features/detect/anomaly.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+### Create anomaly detection rule
+
+Anomaly detection rules contain the configuration information necessary to perform the machine learning analysis. They can run for a specific time period or continuously against incoming data.
+
+For configuration you need to set :
+- a filter query (to aggregate events from a specific area)
+
+![](/assets/xdr/rules/anomaly/172566414-fbb7ecc9-78ea-474d-b1db-f8248fb927c7.png)
+- aggregation method (`Avergae`, `Cardinality`,`Count`, `Max`, `Min`, `Sum`)
```
Comment (aheuze) [inline docs/xdr/features/detect/anomaly.md:27]: Typo here: "Avergae" -> "Average"

Comment (otetard) [reply docs/xdr/features/detect/anomaly.md]: Fixed, thanks!

File: docs/xdr/features/detect/anomaly.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
+# Anomaly Detection Rules
+
+You can use SEKOIA.IO Anomaly Detection Rules to analyse time series data, identify anomalous patterns and raise alerts.
```
Comment (ka0ula) [inline docs/xdr/features/detect/anomaly.md:3]:
```suggestion
You can use SEKOIA.IO Anomaly Detection Rules to analyze time series data, identify anomalous patterns and raise alerts.
```

File: docs/xdr/features/detect/anomaly.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+Anomaly detection rules are based on machine learning anomaly detection features automating the analysis of univariate time series data by creating accurate baselines of normal behavior in your data. These baselines are used to identify anomalous patterns in your time series data and raise alerts with different level of criticality.
```
Comment (ka0ula) [inline docs/xdr/features/detect/anomaly.md:5]:
```suggestion
Anomaly detection rules are based on machine learning anomaly detection features automating the analysis of univariate time series data by creating accurate baselines of normal behavior in your data. These baselines are used to identify anomalous patterns in your time series data and raise alerts with different levels of criticality.
```

File: docs/xdr/features/detect/anomaly.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+This module provide 1-Dimensional time series chart illustrating historical and current data values, forecasting for a specified date range, upper and lower bounds for the expected values, and the anomalies that occur outside these bounds with criticality score.
```
Comment (ka0ula) [inline docs/xdr/features/detect/anomaly.md:7]:
```suggestion
This module provides 1-Dimensional time series chart illustrating historical and current data values, forecasting for a specified date range, upper and lower bounds for the expected values, and the anomalies that occur outside these bounds with criticality score.
```

File: docs/xdr/features/detect/anomaly.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+- Am I interested in monitoring deviation of a specific metric compared to its historical data ?
+- Does my time series have some periodic seasonalities (daily seasonality/weekly seasonlaity) ?
+
+If yes, you can create an anomaly detection rule. An additional tool for specifying if time series data is a good candidate is set to ensure relevancy.
+
+### Create anomaly detection rule
+
+Anomaly detection rules contain the configuration information necessary to perform the machine learning analysis. They can run for a specific time period or continuously against incoming data.
+
+For configuration you need to set :
```
Comment (ka0ula) [inline docs/xdr/features/detect/anomaly.md:23]:
```suggestion
For a complete configuration, you need to set:
```

File: docs/xdr/features/detect/anomaly.md [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+
+![](/assets/xdr/rules/anomaly/77491073/172567318-cb8d9304-b86b-4b2e-ab21-8e43b177324b.png)
+- type of anomalies you want to detect (high, low or both)
+
+![](/assets/xdr/rules/anomaly/77491073/172567367-0aa07dab-e203-441a-8484-2defb9b6189a.png)
+- minimum anomaly score raising alerts (`Low`, `Medium` or `High`)
+- severity of the rule
+
+#### Aggregation methods
+- Count : count events per bucket (for a specific filter or field set to a value)
```
Comment (ka0ula) [inline docs/xdr/features/detect/anomaly.md:43]:
```suggestion
- Count: count events per bucket (for a specific filter or field set to a value)
```

File: docs/xdr/features/detect/anomaly.md [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+- Max (only for numerical data): aggregate max of field values per bucket
+- Average (only for numerical data): aggregate average of field values per bucket
+
+
+### Visualization
+As a result, for a rule that aims to monitor the volume of services installed on systems (query filter: `event.dialect:windows AND action.id:4697`, with a `Count` aggregation method and a `10 min` time interval), we obtain:
+
+![!](/assets/xdr/rules/anomaly/172630813-fc957ee3-0b28-45d1-ace7-696a7208213e.png){: style="max-width:100%"}
+
+Then anomaly identified raise an alert that you can visualize as follows.
```
Comment (ka0ula) [inline docs/xdr/features/detect/anomaly.md:56]:
```suggestion
The anomaly identified raises an alert that you can visualize as follows.
```

**Review by gbossert — APPROVED** [review]

**Pull request merged**